WalletWasabi/WabiSabi/Models/MultipartyTransaction/SigningState.cs

```diff
@@ -6,18 +6,20 @@
 
 namespace WalletWasabi.WabiSabi.Models.MultipartyTransaction
 {
-	public record SigningState(MultipartyTransactionParameters Parameters, ImmutableArray<Coin> Inputs, ImmutableArray<TxOut> Outputs) : IState
+	public record SigningState : MultipartyTransactionState
 	{
-		public ImmutableDictionary<int, WitScript> Witnesses { get; init; } = ImmutableDictionary<int, WitScript>.Empty;
+		public SigningState(ConstructionState state)
+		{
+			Parameters = state.Parameters;
+			Inputs = state.Inputs;
+			Outputs = state.Outputs;
+		}
 
-		public bool IsFullySigned => Witnesses.Count == Inputs.Length;
+		public MultipartyTransactionParameters Parameters { get; }
```

Comment:
can this property be moved to the base class/record instead of being added once per subclass?